### DISCUSSION TITLE
Central managment

### DISCUSSION THREAD

**Discussion by mark-prins** [msupply-foundation/open-msupply#4005] [General]
We plan to refine the purpose of the central server and probably change the UI. 
However, before that, there is a need to administer some 'things' centrally. They don't really fit in the current menu structure.

Here's a proposal:

* change `Admin` to `Settings`, keeping the position and gear icon
* introduce a central-only menu item, called `Manage` / `Management` / `Administration`, which
  * houses centrally managed objects ( in the current example a list of facilities )

Here's a couple of icon suggestions:

<img width="309" alt="Screenshot 2024-05-27 at 12 16 45 PM" src="https://github.com/msupply-foundation/open-msupply/assets/9192912/f30c044f-ee50-48a6-ac75-919582035f99">
<img width="268" alt="Screenshot 2024-05-27 at 12 21 13 PM" src="https://github.com/msupply-foundation/open-msupply/assets/9192912/2eda363e-9b02-43f7-9451-914302878f9e">

Thoughts:
* Not using `Admin` for the new section, even though that's a good name! there is confusion potential
* Someone very pedantic would like all menu items to be nouns, as they are all nouns currently. While `Manage` reads better, it doesn't fit with the other items!
* the menu title `Management` sounds like a boss, and that's not the intention

Thoughts? suggestions?

**Comment by adamdewey** [comment]
Who's the pedant? 😂

I like `Manage` although it is a bit generic - technically the central user would be managing other things too, like the `Catalogue` ?

But maybe it could be a dumping ground for things until they get big / important enough to have their own Nav item?

**Reply by mark-prins** [reply]
I wouldn't be rude enough to label anyone else a pedant 😉 
`Dumping ground` is good - and we can have a little dump truck icon!

**Reply by Chris-Petty** [reply]
Soz for a bit of a ramble but gives some historical context:

I like `Administration`, though in this context it's more or less synonymous with `Manage`.  Fo real though it comes from `Primary` and `central` server being a constant pain for me to explain to new staff, users, and old staff who are confused for the same reason as everyone else - primary and central are synonyms, so invariably they get used interchangeably for the different things they're "meant" to be mean. This became particularly bad gross problem with mirror sync when they became completely different servers. To that end my thoughts a couple years ago was to rename them to concretely what they are:

- Central -> Synchronisation Web Server
- Primary -> Administration Server (or even Administration Site - mirror sync doesn't require the "primary" to use mSupply Server at all!)

I assume the ship has sailed and we are cursed forever in Legacy Desktop though. OMS on the other hand, `Manage` might be better because if anything like "mirror sync" was required on OMS standalone (it's a functional requirement of several current countries...) then we'd have a chance to rename to "Sync Web Server" and "Administration Site", with split responsibilities. If the menu item was `Administration`, it'd get conflated with the "Administration Site". Just as how right now you just have to _know_ what "central" data is configured on which 🙃.

**Reply by adamdewey** [reply]
ooh yeah these are good points @Chris-Petty 

I'd even go so far to say that there's another aspect to consider as there is likely a difference between the people that are managing the system for the national supply chain (and other things - national CCEI management, national EIR??) vs those that are managing the day-to-day operations for the Central Medical Store.

So some countries might want something more like: 

- Central -> National Synchronisation Web Server
- Primary -> National Administration Server (or even Administration Site - mirror sync doesn't require the "primary" to use mSupply Server at all!)
- Remote but hosted centrally -> Central Medical Store

**Reply by Chris-Petty** [reply]
`Remote but hosted centrally -> Central Medical Store` Hahaha I have to say hard no vote to that 😁. Some places Central Medical Store (CMS) is literally the name of the facility that is their well, CMS, NMS whatever one might want to call it. It can be a remote instance at the physical facility. In some cases that instance is the Primary site.

**Reply by adamdewey** [reply]
Ah, sorry, I think this is what I'm trying to highlight though, maybe 🤔

Like, there are a bunch of people that need to operate the stock logistics side of the CMS - so all the distribution, replenishment etc. 

But those same people wouldn't necessarily want to see all the national assets / CCEI, or the national EIR, or perhaps even the Master Data for stock management? 

I guess maybe we could hide / unhide functionality via user preferences if we're careful enough?

**Reply by Chris-Petty** [reply]
I can see the ambiguity that you're raising but don't _understand_ it, I'm not familiar enough with CCEI to really comment I think. Is it because CCEI is currently mixed in the existing UI (i.e. you see the national data and your logged in store data mixed together)?

**Reply by adamdewey** [reply]
Wait, aren't we both saying the same thing? 

- Remote but hosted centrally -> Central Medical Store
- It can be a remote instance at the physical facility. In some cases that instance is the Primary site.

**Reply by adamdewey** [reply]
> I can see the ambiguity that you're raising but don't _understand_ it, I'm not familiar enough with CCEI to really comment I think. Is it because CCEI is currently mixed in the existing UI (i.e. you see the national data and your logged in store data mixed together)?

Yes, that is definitely part of the confusion for me. 

In that case I feel like the user would have 2 roles - looking at their CMS data and looking at the national data.

**Reply by Chris-Petty** [reply]
> Wait, aren't we both saying the same thing?
> 
>     * Remote but hosted centrally -> Central Medical Store
> 
>     * It can be a remote instance at the physical facility. In some cases that instance is the Primary site.

I'm saying some countries call their National Medical Store facility "Central Medical Store" and that it is hosted on premises, while they may have a cloud server where the rest of "Central" mSupply exists.

**Reply by Chris-Petty** [reply]
> > I can see the ambiguity that you're raising but don't _understand_ it, I'm not familiar enough with CCEI to really comment I think. Is it because CCEI is currently mixed in the existing UI (i.e. you see the national data and your logged in store data mixed together)?
> 
> Yes, that is definitely part of the confusion for me.
> 
> In that case I feel like the user would have 2 roles - looking at their CMS data and looking at the national data.

Seems like maybe a nice way of separating out would be good - perhaps below "Cold Chain" could be "National Cold Chain". Probably scope creep for me to suggest though haha. Putting National Cold Chain under "Manage" might be good, up to you if it's a good idea!

**Reply by adamdewey** [reply]
> I'm saying some countries call their National Medical Store facility "Central Medical Store" and that it is hosted on site, while they may have a cloud server where the rest of "Central" mSupply exists.

Ah, yes so for me, this would make sense - it's the scenario when both the primary and the CMS store are on the same instance (same UI) that it gets a bit murky.

Could be gotten around by recommending that the CMS is always installed as a remote site I suppose?

**Reply by adamdewey** [reply]
> Seems like maybe a nice way of separating out would be good - perhaps below "Cold Chain" could be "National Cold Chain". Probably scope creep for me to suggest though haha. Putting National Cold Chain under "Manage" might be good, up to you if it's a good idea!

Ah, I think maybe I was expecting Distribution / Replenishment functionality etc to be removed from the central config server (unsure whether to call it primary or administration or management server now 😂) and only national-level functionality would remain? (i.e. not CMS-level functionality)

**Reply by Chris-Petty** [reply]
Hmm IDK if I quite follow and worried about how github discussions handles more than a dozen replies 😆. My screen keeps jumping around. I'd suggest discussing IRL but rather than commit, first:

- "Manage" is good enough(tm) separation for centrally controlled data right now?

- There is some unclear UX for CCEI (and perhaps things like it later), I agree with my 2c but also keen to step out of going deeper,  that's a CCEI development and budget to use PO and UX resources to figure out?

**Reply by adamdewey** [reply]
Haha, yeah quite a few comments 😆️️️️️️

I'm not sure if we can think of the "Manage" Nav menu like that, as there are (or will be) other centrally controlled data scattered around - e.g. Catalogue > Assets, Cold Chain > Equipment and presumably Catalogue > Items? 

Or are you thinking there should be another Master Data section inside Manage for items etc? 

Ooo there's a thought - maybe we need a Master Data section inside Manage? Or should Manage be called Master Data?!

**Reply by Chris-Petty** [reply]
Yea sorry in my mind all centrally controlled stuff would be under there - Facilities, stores, items, master lists... https://github.com/msupply-foundation/open-msupply/discussions/3723#discussion-6588706

**Reply by Chris-Petty** [reply]
If a store was going to have some control over it's items, e.g. setting a default location of received stock, then that'd be under the usual "store" UI - `Catalogue > Items`

**Reply by adamdewey** [reply]
Ah yeah ok, that would be a lot cleaner for the CMS if it worked like that tbh...

**Comment by roxy-dao** [comment]
I like the first screenshot... but don't feel like it belongs at the bottom of the list.... it should either go at the top of the list or maybe before settings? 🤔

**Reply by jmbrunskill** [reply]
I vote for the sliders icon too.

**Reply by Chris-Petty** [reply]
Just thinking about the top/bottom thing, my intuition from Legacy is that it should be on the bottom because I'd expect all sites to see the UI, and most users to use it infrequently and my preference for infrequenct actions to be at the bottom of the list. With that said though, this will only be displayed to users on the OMS Central Server right? In which case if you're on the central server perhaps administration tasks are most frequent?

I agree with the sliders icon preference.

**Reply by Chris-Petty** [reply]
Though, for just central server you've shifted all of the UI down rather than adding a bonus item at the bottom... I'm back round to preferring bottom haha. My muscle memory for clicking might break if everything shifted!

**Reply by roxy-dao** [reply]
Maybe when we shift to central-actions only central server it can at top? 👀

**Comment by DhanyaHerath** [comment]
Hiiii - days late, and am not sure this is the right place, but is the general direction we're moving in that the central server shouldn't have active stores and therefore should look quite different from this?

**Reply by mark-prins** [reply]
yes! that's the direction I think we should head in. The proposal here is a stop-gap solution really, a way of getting some management functionality into central server for GAPS ahead of the wider redesign / refactor. I would expect central to look different to this eventually.

**Reply by adamdewey** [reply]
My initial thought was to have a policy to never have any active stores on the central server, but I think I'm going back on that now!

For organisations like UNICEF they might have small warehouses managed by 1 or 2 people but effectively sitting at the top of the supply chain and so would want to have store management and central config management in the same interface. 

Could we perhaps make it so that some of the nav items are shown / hidden depending on if there is 1+ active stores on the central site?

**Comment by adamdewey** [comment]
Wondering what the latest consensus is on this?

The upcoming GAPS pilot in December will need all stores to be on a central cloud server, but at the moment that means we end up with central functionality like this showing up for small health facilities: 

<img width="1204" height="1600" alt="image" src="https://github.com/user-attachments/assets/15ee8f24-e40a-44b8-90a5-83b1b7809f56" />

**Comment by roxy-dao** [comment]
There's work upcoming next year for central server ground work, so I don't think it will be scheduled in before then

**Reply by adamdewey** [reply]
Thanks Roxy, we'll have to go with a dual cloud server config then for the GAPS pilot I think
